core/src/main/java/com/sequenceiq/cloudbreak/service/altus/AltusMachineUserService.java

```diff
@@ -176,14 +178,19 @@ public DataBusCredential getOrCreateDataBusCredentialIfNeeded(Long stackId) thro
     }
 
     public CdpAccessKeyType getCdpAccessKeyType(StackDto stackDto) {
-        if (!entitlementService.isECDSABasedAccessKeyEnabled(stackDto.getAccountId())) {
-            return CdpAccessKeyType.ED25519;
-        }
-        Optional<Image> image = componentConfigProviderService.findImage(stackDto.getId());
-        if (image.isEmpty()) {
-            return CdpAccessKeyType.ED25519;
+        if (AwsConstants.AwsVariant.AWS_NATIVE_GOV_VARIANT.variant().value().equals(stackDto.getPlatformVariant())) {
+            Optional<Image> image = componentConfigProviderService.findImage(stackDto.getId());
+            if (image.isEmpty()) {
+                throw new CloudbreakRuntimeException("We can't figure out package versions for image because the image is empty.");
```

Comment:
This error message says nothing about the real issue. If we keep this code  branch we need to say something like this.
"We tried to create ECDSA key type for AWS gov deployment since that's only allowed but were unable to get image package versions since image was undefined."

Not even sure if this is a realistic scenario.